pkg/controller/job/job_controller.go

```diff
@@ -153,7 +153,10 @@ func (jm *JobManager) getPodJob(pod *api.Pod) *experimental.Job {
 		glog.V(4).Infof("No jobs found for pod %v, job manager will avoid syncing", pod.Name)
 		return nil
 	}
-	// TODO: add sorting and rethink the overlaping controllers, internally and with RCs
+	if len(jobs) > 1 {
+		glog.Errorf("user error! more than one job is selecting pods with labels: %+v", pod.Labels)
```

Comment:
As long as the events library in `pkg/client/record/event.go` is used to generate events, you'll get duplicate event coalescing for free. See  [event_compression.md](https://github.com/kubernetes/kubernetes/blob/master/docs/design/event_compression.md)